support-workers/src/main/scala/com/gu/acquisitions/AcquisitionDataRowBuilder.scala

```diff
@@ -149,7 +154,13 @@ object AcquisitionDataRowBuilder {
         case (Collection, Saturday) => VoucherSaturday
         case (Collection, SaturdayPlus) => VoucherSaturdayPlus
         case (Collection, Sunday) => VoucherSunday
-        case _ => VoucherSundayPlus
+        case (Collection, SundayPlus) => VoucherSundayPlus
+        case (NoFulfilmentOptions, _) | (_, NoProductOptions) | (Domestic, _) | (RestOfWorld, _) |
+            (NationalDelivery, EverydayPlus) | (NationalDelivery, SixdayPlus) | (NationalDelivery, WeekendPlus) |
+            (NationalDelivery, SaturdayPlus) | (NationalDelivery, SundayPlus) =>
+          throw new Exception(
+            s"Unrecognised (fulfilmentOptions, productOptions) combination: ($fulfilmentOptions, $productOptions)",
+          )
```

Comment:
Is `Exception` the right exception type to use here?

Reply:
I guess an alternative here would be to add a new case object `InvalidPrintProduct` and use that here: then at least the acquisitions would make it to the DB ok (and nothing would crash). It would mean we could have quiet failures though.

Reply:
if you want to do something in response to the exception, it makes sense to have a proper type, otherwise I generally just throw a built in (usually runtime exception).  Although from a "functional programming" point of view, catching exceptions is frowned upon, so it depends how far away from FP the surrounding code is.

Reply:
At the moment there isn’t anything I want to do in response to the exception, as this case shouldn’t actually happen: I just want it to be a bit more visible than it was. At the moment incorrect combinations all go in as VoucherSundayPlus, which happened when I added the new fulfilment option: explicitly matching on all the cases means adding a new fulfilment option without updating this code gives a compiler error, which makes it harder to miss, but raises the question of what to do in these cases.

As of this morning, I’ve gone with an `InvalidPrintProduct` case class, which means nothing will crash with an unhandled exception and it should be clear from the acquisition data when something’s gone wrong.